internal/store/cluster_installation_test.go

```diff
@@ -564,3 +565,269 @@ func TestDeleteClusterInstallations(t *testing.T) {
+		},
+	}
+	err := sqlStore.CreateGroup(group1)
+	require.NoError(t, err)
+	groupID1 := group1.ID
+
+	time.Sleep(1 * time.Millisecond)
+
+	annotations := []*model.Annotation{{Name: "annotation1"}, {Name: "annotation2"}}
+
+	installation1 := &model.Installation{
+		OwnerID:   ownerID1,
+		Version:   "version",
+		DNS:       "dns.example.com",
+		Database:  model.InstallationDatabaseMysqlOperator,
+		Filestore: model.InstallationFilestoreMinioOperator,
+		Size:      mmv1alpha1.Size100String,
+		Affinity:  model.InstallationAffinityIsolated,
+		GroupID:   &groupID1,
+		CRVersion: model.V1betaCRVersion,
+		State:     model.InstallationStateCreationRequested,
+	}
+
+	err = sqlStore.CreateInstallation(installation1, annotations)
+	require.NoError(t, err)
+	time.Sleep(1 * time.Millisecond)
+
+	installation2 := &model.Installation{
+		OwnerID:   ownerID1,
+		Version:   "version2",
+		Image:     "custom-image",
+		DNS:       "dns2.example.com",
+		Database:  model.InstallationDatabaseMysqlOperator,
+		Filestore: model.InstallationFilestoreMinioOperator,
+		Size:      mmv1alpha1.Size100String,
+		Affinity:  model.InstallationAffinityIsolated,
+		GroupID:   &groupID2,
+		CRVersion: model.DefaultCRVersion,
+		State:     model.InstallationStateStable,
+	}
+
+	err = sqlStore.CreateInstallation(installation2, nil)
+	require.NoError(t, err)
+
+	time.Sleep(1 * time.Millisecond)
+
+	sourceClusterID := model.NewID()
+	clusterInstallation1 := &model.ClusterInstallation{
+		ClusterID:      sourceClusterID,
+		InstallationID: installation1.ID,
+		Namespace:      "namespace_10",
+		State:          model.ClusterInstallationStateCreationRequested,
+	}
+
+	time.Sleep(1 * time.Millisecond)
+
+	targetClusterID := model.NewID()
+	clusterInstallation2 := &model.ClusterInstallation{
+		ClusterID:      sourceClusterID,
+		InstallationID: installation2.ID,
+		Namespace:      "namespace_11",
+		State:          model.ClusterInstallationStateCreationRequested,
+	}
+
+	err = sqlStore.CreateClusterInstallation(clusterInstallation1)
+	require.NoError(t, err)
+
+	err = sqlStore.CreateClusterInstallation(clusterInstallation2)
+	require.NoError(t, err)
+
+	clusterInstallations, err := sqlStore.GetClusterInstallations(&model.ClusterInstallationFilter{Paging: model.AllPagesNotDeleted(), ClusterID: sourceClusterID})
+	require.NoError(t, err)
+
+	oldCIsIDs := getClusterInstallationIDs(clusterInstallations)
+	err = sqlStore.MigrateClusterInstallations(clusterInstallations, targetClusterID)
+	require.NoError(t, err)
+
+	var installations []string
+	installations = append(installations, installation1.ID)
+	installations = append(installations, installation2.ID)
+	newCIsIDs := getClusterInstallationIDs(clusterInstallations)
+	var installationIDs []string
+	for _, ci := range clusterInstallations {
+		installationIDs = append(installationIDs, ci.InstallationID)
+	}
+	err = sqlStore.SwitchDNS(oldCIsIDs, newCIsIDs, installationIDs)
+	require.NoError(t, err)
```

Comment:
I think it might be worth it to do a quick check after that, that everything switched correctly be fetching the resources and checking the values that should change.

Reply:
Added some verification steps.